test/Polly.Specs/Wrap/PolicyWrapContextAndKeyAsyncSpecs.cs

```diff
@@ -296,11 +298,7 @@ public async Task Should_restore_PolicyKey_of_outer_policy_to_execution_context_
 
         await policyWrap.ExecuteAsync(async () => await Task.Run(() => // Regression test for issue 510
         {
-            throw new Exception();
-#pragma warning disable 0162 // unreachable code detected
-            return ResultPrimitive.WhateverButTooLate;
-#pragma warning restore 0162
-
+            return ThrowException();
```

Comment:
Does this preserve the exact scenario in issue #510 that's being tested?

If not, then the existing suppression was correct.

Reply:
This should preserve the exact scenario, to make sure that I have added the same functional structure as before